lms/djangoapps/instructor/utils.py

```diff
@@ -36,3 +39,120 @@ def get_module_for_student(student, usage_key, request=None):
     descriptor = modulestore().get_item(usage_key, depth=0)
     field_data_cache = FieldDataCache([descriptor], usage_key.course_key, student)
     return get_module(student, request, usage_key, field_data_cache)
+
+
+def collect_ora2_data(course_id):
+    """
+    Query MySQL database for aggregated ora2 response data.
+    """
+    cursor = get_read_replica_cursor_if_available(db)
+
+    #Syntax unsupported by other vendors such as SQLite test db
+    if db.connection.vendor != 'mysql':
+        return '', ['']
+
+    raw_queries = ora2_data_queries().split(';')
+
+    cursor.execute(raw_queries[0])
+    cursor.execute(raw_queries[1], [course_id])
+
+    header = [item[0] for item in cursor.description]
+
+    return header, cursor.fetchall()
+
+
+def ora2_data_queries():
+    """
+    Wraps a raw SQL query which retrieves all ORA2 responses for a course.
+    """
+
+    # pylint: disable=invalid-name
+    RAW_QUERY = """
```

Comment:
That sounds reasonable. I'm going to ask @stephensanchez to weigh in here. Steve, could you just look at the one question posted above? Usman and his team are going to look at this PR in their sprint this week, but this seems like a bit more of an architectural question.